src/Integration.Vsix/CFamily/CMake/CompilationDatabaseRequest.cs

```diff
@@ -78,7 +78,7 @@ public void WriteRequest(BinaryWriter writer)
             WriteHeader(writer);
 
             // Required inputs
-            WriteSetting(writer, "File", databaseEntry.File);
+            WriteSetting(writer, "File", Context.File);
```

Comment:
We need to write the analyzed header file, not the code entry -- the code entry is used only for the build parameters. I added a test for this.